src/tds/net.c

```diff
@@ -669,6 +669,8 @@ tds_select(TDSSOCKET * tds, unsigned tds_sel, int timeout_seconds)
 
 			switch (sock_errno) {
 			case TDSSOCK_EINTR:
+			case EAGAIN:
+			case TDSSOCK_EINPROGRESS:
```

Comment:
`pool` does not return these errors as far as I know

Reply:
I don't know what Sybase libraries do in this scenario; having retired our last Solaris systems years ago, we now have them only on Linux, which doesn't document either error code as a possible `poll` outcome.  However, both error codes generally call for retrying rather than bailing, and POSIX says `poll` can yield `EAGAIN`: https://www.unix.com/man-page/posix/3p/poll/.  That said, a closer look indicates that the subsequent assertion will need to be updated accordingly; sorry for missing that earlier.